### ISSUE TITLE
Error on latest version on Jest (23.5.0)

### ISSUE THREAD

**Issue by JPStrydom** [clarkbw/jest-localstorage-mock#88]
Here's a checklist you can use to make sure your issue can be more easily resolved:

- [x] Check against the [latest](https://github.com/clarkbw/jest-localstorage-mock/releases/latest)
- [x] Include a link to the example application source code
- [x] Paste into this issue the exact error messages messages you're seeing

Whenever I run my tests with Jest version 21.2.1 everything works fine, but when I install Jest version 23.5.0 I receive the following error:
```
Error: expect(jest.fn())[.not].toHaveBeenCalledWith()
jest.fn() value must be a mock function or spy.
Received:  function: [Function getItem]
```
i.e. `localStorage.getItem` is not being mocked out

I'm on the latest version of jest-localstorage-mock (2.2.0) and my test setup file looks as follows:

```
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-localstorage-mock';

configure({ adapter: new Adapter() });

beforeEach(() => {
    localStorage.clear();
});

```

**Comment by JPStrydom** [comment]
It seems that upgrading to Jest 22.0.0 or beyond causes the issue

**Comment by JPStrydom** [comment]
The following can be used as a work around:

```
class LocalStorageMock {
    constructor() {
        this.__STORE__ = {};
    }

    clear = jest
        .fn()
        .mockName('localStorage.clear')
        .mockImplementation(() => (this.__STORE__ = {}));

    getItem = jest
        .fn()
        .mockName('localStorage.getItem')
        .mockImplementation(key => this.__STORE__[key] || null);

    setItem = jest
        .fn()
        .mockName('localStorage.setItem')
        .mockImplementation((key, value) => (this.__STORE__[key] = value.toString()));

    removeItem = jest
        .fn()
        .mockName('localStorage.removeItem')
        .mockImplementation(key => delete this.__STORE__[key]);
}

Object.defineProperty(window, 'localStorage', {
    value: new LocalStorageMock()
});

beforeEach(() => {
    window.localStorage.clear();
    window.localStorage.clear.mockClear();
    window.localStorage.getItem.mockClear();
    window.localStorage.setItem.mockClear();
    window.localStorage.removeItem.mockClear();
});

```

Just add this code in your setup file 👍

**Comment by th3fallen** [comment]
```
jest.spyOn(Storage.prototype, 'setItem');
  jest.spyOn(Storage.prototype, 'getItem');
  jest.spyOn(Storage.prototype, 'removeItem');
``` 
Fixes it for me per 
https://github.com/facebook/jest/issues/6858
and https://github.com/facebook/jest/issues/6798

**Comment by JPStrydom** [comment]
I'll give it a try. But the we still can't access the mocked `__STORE__` variable unfortunately.

**Comment by clarkbw** [comment]
Seems like this is related 
https://github.com/clarkbw/jest-localstorage-mock/pull/80

**Comment by AlexandrLi** [comment]
Hey, guys, I guess this issue could be closed due merge of #80. Btw, I don't see any release with this changes. Any comments on this?
